vlib/v/checker/tests/match_last_branch_only_casted.out

```diff
@@ -0,0 +1,7 @@
+vlib/v/checker/tests/match_last_branch_only_casted.vv:6:10: error: the type of the last expression of the first match branch was `u8`, which is not compatible with `int literal`
+    4 |         'a' { 10 }
```

Comment:
If the first branch was `u8`, and the `else` is `u8`, then they _are_ compatible.  This shouldn't give an error.

Reply:
Oh, I meant to write in the other way.

Reply:
I agree with @JalonSolov, the type names in the error message should be swapped in general.

There is a more specific problem here too though.
The first branch's last expression value is an integer literal. Absent more precise type, integer literals, should be treated as having the type `int`, which is wide enough to contain values of type `u8` too (0..255 fits inside an int without a problem).

In other words, do swap the type names, but make it *also*, so that the checker error is not triggered at all for that specific example.

Reply:
Done